tests/platform_tests/cli/test_show_platform.py

```diff
@@ -336,6 +336,47 @@ def test_show_platform_fan(duthosts, enum_supervisor_dut_hostname):
                   " No Fans are displayed with OK status on '{}'".format(duthost.hostname))
 
 
+def check_show_platform_sensor_output(cmd, duthost):
+    """
+    @summary: Run and verify output of `show platform [voltage|current]`. Expected output
+              is "Sensor Not detected" or a table of sensor status data with 8 columns.
+    """
+    num_expected_clos = 8
+
+    logging.info("Verifying output of '{}' on '{}'...".format(cmd, duthost.hostname))
+    raw_output_lines = duthost.command(cmd)["stdout_lines"]
+
+    pytest_assert(len(raw_output_lines) > 0, "There must be at least one line of output on '{}'".format(hostname))
+    if len(raw_output_lines) == 1:
+        pytest_assert(raw_output_lines[0].strip() == "Sensor Not detected",
+                      "Unexpected sensor status output on '{}'".format(hostname))
+    else:
+        pytest_assert(len(raw_output_lines) > 2,
+                      "There must be at least two lines of output if any sensor is detected on '{}'".format(hostname))
+        second_line = raw_output_lines[1]
+        field_ranges = util.get_field_range(second_line)
+        pytest_assert(len(field_ranges) == num_expected_clos, "Output should consist of {} columns on '{}'".
+                      format(num_expected_clos, hostname))
```

Comment:
Can this also check that the alarm status for every line of sensor data reads false, indicating no alarm? Perhaps `test_thermal.py` can instead be copied/expanded for coverage of sensors of other types (voltage, current, etc.)

Reply:
API tests should be in addition to this request (I'm working on that piece now). Not sure it makes sense for API tests to be validating the thresholds against the operational values of the sensors, just that they all return values of the correct type, so it would make sense to verify the alarm status is false for all sensors through the CLI here.